server/tserver/src/main/java/org/apache/accumulo/tserver/compactions/ExternalCompactionJob.java

```diff
@@ -76,7 +76,7 @@ public TExternalCompactionJob toThrift() {
 
     List<InputFile> files = jobFiles.entrySet().stream().map(e -> {
       var dfv = e.getValue();
-      return new InputFile(e.getKey().getNormalizedPathStr(), dfv.getSize(), dfv.getNumEntries(),
+      return new InputFile(e.getKey().getMetadata(), dfv.getSize(), dfv.getNumEntries(),
```

Comment:
As a follow on it would be good to have an IT that forces an external compaction on a tablet that has files with ranges.  Construct the situation such that if the external compaction does not honor the ranges it would end up creating a file with more data than it should have.